### PR TITLE
Reworking makeshift explosives part 1

### PR DESCRIPTION
**Pull request by NeviNovat**
#### Summary

```SUMMARY: Content "Reworking makeshift explosives part 1."```

#### Purpose of change
There is a lack of makeshift explosives one can make. I aim to add explosives that are auto-learned for every fab level from 0 to 5 that are different and might still be used once the player has a bigger level.
This should take the pressure off the pipe bomb and the makeshift grenade to perform and allow them to be balanced or made harder to get, since new more simpler and less powerful explosives will be available.

#### Describe the solution
Add more makeshift explosives and balance them with existing ones.
A total of 8 new makeshift explosives and reworking of 3 makeshift explosives is planed.
This PR ( part 1) adds makeshift explosive for lv 0-1 and adds new items that will help in the future.

#### Describe alternatives you've considered
Add less explosives making a single line of apparent progression.

#### Additional context
I play-tested the modified items for about 2-3 hours (on and off) and those values assigned seem to vary wildly because, form my understanding , the calculations always assume the shrapnel to be a spherical ball ( and the ball can only be 0.1g in minimum weight. ) so I look at the damage done not the numbers. Even so I think I'll change them in the future.

The match head powder added is similar to blackpowder from what I understand that's why I added it as an alternative, I saw a video were someone reloaded some ammo with the stuff and did some tests but I can't find it ( the video showed that match powder can achieve 2/3 the velocity of the same amount of smokeless if I remember right , this is kinda the same for black powder thats why the value were similar ). In any way right now the match head powder primary role is to be an alternative while making fuses.

The **overall idea** is to add 2 apparent lines one that focuses on small but numerous shrapnel and the other with bigger less numerous fragments this adds a choice to the player. Because the one with smaller shrapnel is more suited for damaging opponents more consistently and does poor when it comes in contact with armored opponents, while the other one is great at thinning the number of targets but can vary wildly with the spread sometimes not damaging opponents at all but can deal with armor a lot better ( can be good for bigger opponents ).

I don't understand **how to fix this error**:
" src/item_group.cpp:121 [virtual void Single_item_creator::check_consistency() const] item id matchbomb is unknown
22:47:44.940 ERROR : src/item_group.cpp:121 [virtual void Single_item_creator::check_consistency() const] item id matchbomb is unknown "

This error doesn't affect the game in any way I can tell and I **can't find it**.

If there is a need for an better explanation for some items or crafting recipe I'll provide. Also I can remove items or recipe if needed **BUT** I prefer not to, I'll rather change them.

- I'll change the description of the PR over time if I missed something or added something.

### CODE REVIEW

**Comment by kevingranade** [comment]
I know we already have sand bombs, but I can't find any sources that indicate that sand or glass provides a meaningfully deadly fragment source, if you want to add these, please provide some sources indicating that this is something that is done. 
Also it doesn't seem like a worthwhile thing to do, since scrap metal is so easy to scavenge.

**Comment by NeviNovat** [comment]
> I know we already have sand bombs, but I can't find any sources that indicate that sand or glass provides a meaningfully deadly fragment source, if you want to add these, please provide some sources indicating that this is something that is done.
> Also it doesn't seem like a worthwhile thing to do, since scrap metal is so easy to scavenge.

I don't know of any examples and I only done this because items with sand and glass were used ( the nail bomb, RDX sand bomb and the match head bomb ( match head bomb is currently not working I believe ) ) and tried to preserve what was already there, since I was believing that this is the way something like this is usually done ( #32275 were the nail bomb was fixed someone said that adding items and not removing existing ones is a way to resolve a possible issue )  . I have doubts about the effectiveness of such weapon and I think that the recipe and the ideas for the sand explosives and glass containers being used is old and forgotten, and I am ok with removing them. That would mean removing the : glass bomb , sand bomb and RDX sand bomb.

So in short: I only added the sand bomb and changed the match head bomb to be the glass bomb since similar item already existed. I am doubtful of the ability of such bombs being deadly thats why I have them be less damaging and for use in softening up zombies ( but i still can't get them not to kill in close proximity and still be somewhat useful ). I have no problem removing the items that are made of glass or use sand as primary shrapnel.

**Comment by kevingranade** [comment]
That's the only issue I see, other than that yea harvesting match heads < gunpowder < smokeless powder < more exotic explosives.

**Comment by NeviNovat** [comment]
> That's the only issue I see, other than that yea harvesting match heads < gunpowder < smokeless powder < more exotic explosives.

I am thinking about having different items for different explosives used in crafting but the item names would look like black powder makeshift grenade, smokeless powder makeshift grenade, RDX makeshift grenade, etc. So I am looking at better more short names to name them, but if I don't come up with something that's ok I'll just use what I got.

**Comment by kevingranade** [comment]
Need to remove the the matchbomb item_group entries and move the item definition to data/json/items/obsolete.json

**Comment by NeviNovat** [comment]
Ty for helping, the matchbomb is not in any itemgroups from my search.

**Comment by kevingranade** [comment]
This pull request has been mentioned on **Cataclysm: Dark Days Ahead**. There might be relevant details there:

https://discourse.cataclysmdda.org/t/im-getting-an-error-in-he-newest-experimental/21610/2
